app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class UsersController < ApplicationController
+      # POST /api/v1/users.json (the json format is required for an explicit API)
+      # Expects:
+      #     params with the fallowing format: { user: { :name, :email, :password, :password_confirmation } }
+      # Does:
+      #     - Filters and validates the recieved params( throws ActionController::ParameterMissing for missed params).
+      #     - Creates and saves a new user record in Users table with the provided parameters.
+      #     - Returns json with the fallowig format : { msg: String, data: Array[serializable objects]
+      #      ,errors: Array[String] }
+      #     and status_code: 201 (created) | 400 (bad request).
+      #         msg = response_msgs[:success] and errors = [] for successful save to db with status_code = 201.
+      #         msg = response_msgs[:failed] and errors = user.errors.to_a for unsuccessfuly save to db
+      #         with status_code = 400.
```

Comment:
Let's simplify this a bit. 2 or 3 lines simply explaining what this does, the params it accepts and what it returns should be good

Reply:
Alright,

Reply:
Done,